### DISCUSSION TITLE
WARNING: Could not open directory '/Application/KiCad/KiCad.app/Contents/SharedSupport/symbols' @ [/Users/victorezekiel/Documents/fsae/working_bspd/<stdin>:1]

### DISCUSSION THREAD

**Discussion by ibkvictor** [xesscorp/skidl#178] [Q&A]
Hi

`Skidl` is unable to locate my symbol dir. I have use the `export` command following instructions in the documentation. However, I still get the errors below. Thanks in advance for the assistance.

Platform: Mac OS , kicad 6

```
>>> from skidl import *
WARNING: KICAD_SYMBOL_DIR environment variable is missing, so the default KiCad symbol libraries won't be searched. @ [/Users/victorezekiel/Documents/fsae/working_bspd/<frozen importlib._bootstrap_external>:783=>/Users/victorezekiel/Documents/fsae/working_bspd/<frozen importlib._bootstrap>:219]
>>> lib_search_paths
lib_search_paths
>>> lib_search_paths
lib_search_paths
>>> lib_search_paths[KICAD]
['.']
>>> lib_search_paths[KICAD] = ["/Application/KiCad/KiCad.app/Contents/SharedSupport/symbols"]
>>> search("74HC4060D")
WARNING: Could not open directory '/Application/KiCad/KiCad.app/Contents/SharedSupport/symbols' @ [/Users/victorezekiel/Documents/fsae/working_bspd/<stdin>:1]
```

**Comment by koob1018** [comment]
hi
have you solved?
i faced this problem too...

**Reply by devbisme** [reply]
It really helps when trying to solve a problem if you submit the information below. You can also try the development version of SKiDL to see if that will cure the problem:
```bash
pip install git+https://github.com/devbisme/skidl@development
```

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS [type & version]
 - Python version
 - SKiDL version [e.g. 22]

**Additional context**
Add any other context about the problem here.

**Comment by ibkvictor** [comment]
Thanks for the info @devbisme 

@koob1018. I solved it manually (directly altering skidl and placing my dir there), but there are two ways to go about it:

1: As provided in the documentation
[ .] Locate the exact /path/to/bin of your kicad python distribution
[ .] create a `venv` environment or one might already exist
[ .] install skidl and all required packages
[ .] set the symbol directory using: `export KICAD_SYMBOL_DIR="/usr/share/kicad/library`

2: Manually by cloning and directly adding your file path
Clone the skidl package and add your path/to/symbol_dir into the `skidl.py` file
You should add it somewhere around here.

```
# If no configuration files were found, set some default lib search paths.
if "lib_search_paths" not in skidl_cfg:
    skidl_cfg["lib_search_paths"] = {tool: ["."] for tool in ALL_TOOLS}

    # Add the location of the default KiCad part libraries.
    try:
        skidl_cfg["lib_search_paths"][KICAD].append(os.environ["KICAD_SYMBOL_DIR"])
    except KeyError:
        active_logger.warning(
            "KICAD_SYMBOL_DIR environment variable is missing, so the default KiCad symbol libraries won't be searched."
        )

    # Add the location of the default SKiDL part libraries.
    default_skidl_libs = os.path.join(
        os.path.dirname(os.path.abspath(__file__)), "libs"
    )
    skidl_cfg["lib_search_paths"][SKIDL].append(default_skidl_libs)
```
Hope this helps :)
and sorry for the late response.
